tests/test_hoist.py

```diff
@@ -0,0 +1,264 @@
+"""
+A selection of tests for the generic hoist functionality.
+"""
+from pathlib import Path
+import pytest
+import numpy as np
+
+from conftest import available_frontends, jit_compile, clean_test
+from loki import FindVariables
+from loki.expression import symbols as sym
+from loki import ir, as_tuple, is_iterable
+from loki import Scheduler
+from loki.transform import HoistAnalysis, HoistSynthesis
+
+write_transformation_output = False
+
+
+@pytest.fixture(scope='module', name='here')
+def fixture_here():
+    return Path(__file__).parent
+
+
+@pytest.fixture(name='config')
+def fixture_config():
+    """
+    Default configuration dict with basic options.
+    """
+    return {
+        'default': {
+            'mode': 'idem',
+            'role': 'kernel',
+            'expand': True,
+            'strict': True,
+        },
+        'routines': []
+    }
+
+
+def compile_and_test(scheduler, here, a=(5,)):
```

Comment:
I would suggesting here an additional argument with something like `test_name` that is injected into the file names. That avoids the risk that leftover files from a different test are used.